src/serverConfigParamsUtils.ts

```diff
@@ -0,0 +1,114 @@
+import Joi, { CustomHelpers } from 'joi';
+import { BigNumber, utils, Wallet } from 'ethers';
+
+const MAX_POSSIBLE_PORT_NUMBER = 65535;
+
+const isBigNumberish: Joi.CustomValidator = (
+  value: string,
+  helpers: CustomHelpers
+) => {
+  //Some values are automatically translated to their numeric representation (i.e. 1e3 to 1000)
+  //Here we need to put them back to their original representation, since strings like 1e3 are invalid
+  //BigNumber strings and fails at runtime
+  value = helpers.original as string;
+
+  //Taken directly from ethers library https://github.com/ethers-io/ethers.js/blob/f97b92bbb1bde22fcc44100af78d7f31602863ab/packages/bignumber/src.ts/bignumber.ts#L28
+  const isValidBigNumberRepresentation =
+    value != null &&
+    (BigNumber.isBigNumber(value) ||
```

Comment:
I would remove this, this could be misleading, I cannot set a isBigNumberish property as bigNumber, therefore I don't think that we should validate it.